docs/source/contributing/developing.adoc

```diff
@@ -280,6 +280,17 @@ For example, tests can be run against a stopped {project} instance with the imag
 $ make integration_all RUN_BEFORE_FEATURE="start; stop; config set image-caching true"
 ----
 
+COPY_OC_FROM::
+
+To test {project} with a non-default `oc` binary you can use the `COPY_OC_FROM` parameter.
+This parameter specifies the path to the `oc` binary which will be copied into the `$MINISHIFT_HOME` folder before each feature run.
+Note that the `oc` binary must be contained in the same directory structure which {project} uses for the storage of `oc` binaries, for example: `<version>/<platform>/oc(.exe)`.
+For example, to test {project} with locally stored `oc v3.7.23` run:
+
+----
+$ make integration COPY_OC_FROM=~/custom-oc/v3.7.23/linux/oc RUN_BEFORE_FEATURE="config set openshift-version v3.7.23"
+----
+
```

Comment:
LGTM, however if we add a note like for mac user use path COPY_OC_FROM=~/custom-oc/v3.7.23/darwin/oc and for win user the path would be COPY_OC_FROM=~/custom-oc/v3.7.23/win/oc or can be rephrased in a better way. This would really increase the doc experience to some extent cc_ @robin-owen

Reply:
@amitkrout Good point. Especially for macOS users, "darwin" is not obvious as the platform name. We could use a short explanation of which directory names are used for each platform.

Reply:
> Especially for macOS users, "darwin" is not obvious as the platform name.

We might then consider adding a short description of `MINISHIFT_HOME` directory structure into documentation, it might help some people to orient better in it and encourage them to be check it as part of the debugging/issue raising. However I do not think the integration tests passage is the right part to explain this as it does not seem systematic and logical to me.

I would suggest to change the command example to: `$ make integration COPY_OC_FROM=<path-to-oc-directory>/v3.7.23/<platform>/oc RUN_BEFORE_FEATURE="config set openshift-version v3.7.23"`. Would that be better? WDYT? @robin-owen @amitkrout

Reply:
> We might then consider adding a short description of MINISHIFT_HOME directory structure into documentation, it might help some people to orient better in it and encourage them to be check it as part of the debugging/issue raising. However I do not think the integration tests passage is the right part to explain this as it does not seem systematic and logical to me.

Agree, however when as user i see public repo of oc v3.7.23 [1] for mac, the URL confuses me to select the platform for COPY_OC_FROM. At first look I personally feel here the platform is not darwin atleast. 

[1] https://mirror.openshift.com/pub/openshift-v3/clients/3.7.23/macosx/ 

> I would suggest to change the command example to: $ make integration COPY_OC_FROM=<path-to-oc-directory>/v3.7.23/<platform>/oc RUN_BEFORE_FEATURE="config set openshift-version v3.7.23". Would that be better? WDYT

As earlier as i mentioned we can rephrase it in a better way. This looks more logical and we can make it more precises like
```$ make integration COPY_OC_FROM=<path-to-oc-directory>/v3.7.23/<linux or windows or darwin>/oc RUN_BEFORE_FEATURE="config set openshift-version v3.7.23"```

make sense ?

Reply:
@amitkrout makes sense, implemented.